include/aikido/control/ros/util.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef AIKIDO_CONTROL_ROS_UTIL_HPP_
+#define AIKIDO_CONTROL_ROS_UTIL_HPP_
+#include <chrono>
+#include <mutex>
+#include <ros/ros.h>
+#include <ros/callback_queue.h>
+#include <actionlib/client/action_client.h>
+
+namespace aikido {
+namespace control {
+namespace ros {
+
+template <class ActionSpec, class TimeoutDuration, class PeriodDuration>
+bool waitForActionServer(
+  actionlib::ActionClient<ActionSpec>& actionClient,
+  ::ros::CallbackQueue& callbackQueue,
+  TimeoutDuration timeoutDuration = std::chrono::milliseconds{ 1000 },
+  PeriodDuration periodDuration = std::chrono::milliseconds{ 10 }
+)
+{
+  using Clock = std::chrono::steady_clock;
+
+  const auto startTime = Clock::now();
+  const auto endTime = startTime + timeoutDuration;
+  auto currentTime = startTime + periodDuration;
```

Comment:
I based it off this : 
https://github.com/personalrobotics/aikido/blob/master/src/control/ros/RosTrajectoryExecutor.cpp#L157
Can you help me figure it out? @gilwoolee any comments?